src/Activity.php

```diff
@@ -32,4 +34,16 @@ public function subscribe(Visitor $visitor, Membership $subscription): array
         }
         return $activityVisits;
     }
+    public function getCalendar(\DateTime $startTime, \DateInterval $period): array
+    {
+        $calendar = [];
+        foreach ($this->schedule->materialize($startTime, $period) as $day) {
+            $activity = new \stdClass();
+            $activity->time = $day;
+            $activity->activity = $this->name;
+            $activity->leader = $this->leader;
+            array_push($calendar, $activity);
+        }
+        return $calendar;
+    }
```

Comment:
Let's encapsulate the calendar logic in `TimeTable` object. To do so you need to extend the public interface of this class.